### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by IvanLuchkin**
Authentication implementation completed

### CODE REVIEW

File: src/main/java/cinema/Main.java [hunk 1]
```diff
@@ -44,5 +49,9 @@ public static void main(String[] args) {
         System.out.println("Requested date: " + requestedDate);
         List<MovieSession> res = movieSessionService.findAvailableSessions(1L, requestedDate);
         System.out.println(res);
+
+        User newUser = authService.register("test@gmail.com", "1234");
```
Comment (Max-Pochepets) [inline src/main/java/cinema/Main.java:53]: what if we create new user within the same email?

File: src/main/java/cinema/security/util/Pbkdf2PasswordEncoder.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+@Service
+public class Pbkdf2PasswordEncoder implements PasswordEncoder {
+    @Override
+    public String getHash(String password, byte[] salt) {
+        KeySpec spec = new PBEKeySpec(password.toCharArray(), salt, 65536, 128);
+        try {
+            SecretKeyFactory factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
```
Comment (Max-Pochepets) [inline src/main/java/cinema/security/util/Pbkdf2PasswordEncoder.java]: lets make the name of the algorithm a constant variable

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/Main.java (hunk 1 above)
Comment (IvanLuchkin) [reply src/main/java/cinema/Main.java:53]: Weird question, an exception will be thrown

Comment (Max-Pochepets) [reply src/main/java/cinema/Main.java:53]: I mean it would be nice to handle such things. Up to you

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/Main.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/cinema/Main.java:53]: main method will throw an exception. Such exception will be handled in other places (controllers for example). Here we can leave it as it is. But you are right, In general, we should handle such situations

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

**Pull request merged**